toolchain/driver/driver.cpp

```diff
@@ -159,29 +162,39 @@ auto Driver::RunDumpSubcommand(DiagnosticConsumer& consumer,
     return false;
   }
 
+  CARBON_VLOG() << "*** TokenizedBuffer::Lex ***\n";
   auto tokenized_source = TokenizedBuffer::Lex(*source, consumer);
+  CARBON_VLOG() << "*** TokenizedBuffer::Lex done ***\n";
   if (dump_mode == DumpMode::TokenizedBuffer) {
+    CARBON_VLOG() << "Finishing output.";
     consumer.Flush();
     tokenized_source.Print(output_stream_);
     return !tokenized_source.has_errors();
   }
+  CARBON_VLOG() << tokenized_source;
 
+  CARBON_VLOG() << "*** ParseTree::Parse ***\n";
   auto parse_tree = ParseTree::Parse(tokenized_source, consumer, vlog_stream_);
+  CARBON_VLOG() << "*** ParseTree::Parse done ***\n";
   if (dump_mode == DumpMode::ParseTree) {
     consumer.Flush();
     parse_tree.Print(output_stream_, parse_tree_preorder);
     return !tokenized_source.has_errors() && !parse_tree.has_errors();
   }
+  CARBON_VLOG() << parse_tree;
```

Comment:
For the parse tree, IR, & tokenized_source: are the beginning & end, and the content of the dump already clear enough to know what's shown? Or would it help to have an explicit header/footer to make sense of it?
Like
```suggestion
  CARBON_VLOG() << "ParseTree dump:";
  CARBON_VLOG() << parse_tree;
  CARBON_VLOG() << "ParseTree end";
```

Reply:
In output, this'll be immediately after "ParseTree::Parse done", and will be bracketed with "SemanticsIR::MakeFromParseTree"; I was leaning away from extra bracketing because it felt more redundant (if you felt strongly, I'd probably just make the "done" bracket more conditional, and change the text). The dumps also have a familiar structure to them.

Reply:
Understood, that seems acceptable then, thank you! ✅

Reply:
Also see #2446 on this